compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -844,7 +844,7 @@ private void createRecordField(MethodVisitor mv, BField field) {
         loadType(mv, field.type);
 
         // Load field name
-        mv.visitLdcInsn(field.name.value);
+        mv.visitLdcInsn(decodeIdentifier(field.name.value));
```

Comment:
since we use `putValue()` generated method to initialize fields, this change will not break record value creation. 
+1